Scripts/Utilities/ObjectUtils.cs

```diff
@@ -200,6 +200,34 @@ public static Component AddComponent(Type type, GameObject go)
             return component;
         }
 
+        public static T CopyComponent<T>(T sourceComponent, GameObject targetGameObject) where T : Component
```

Comment:
Check out EditorUtility.CopySerialized. You should be able to just AddComponent, and then CopySerialized(source, target)

Reply:
good call; I'm refactoring the method now